src/components/KPrompt/KPrompt.vue

```diff
@@ -1,332 +1,189 @@
+    type: String as PropType<ButtonAppearance>,
+    default: 'primary',
+  },
+  actionButtonDisabled: {
+    type: Boolean,
+    default: false,
   },
   cancelButtonText: {
     type: String,
     default: 'Cancel',
   },
-  /**
-   * Boolean to disable action buttons while a submission is occurring. Display
-   * spinner on action button.
-   */
-  actionPending: {
-    type: Boolean,
-    default: false,
+  cancelButtonAppearance: {
+    type: String as PropType<ButtonAppearance>,
+    default: 'tertiary',
   },
-  isVisible: {
+  cancelButtonDisabled: {
     type: Boolean,
     default: false,
   },
-  /**
-   * The max-height of the prompt. **Default: `400px`**.
-   */
-  maxHeight: {
-    type: String,
-    required: false,
-    default: '400px',
-  },
-  /**
-   * Use this prop to require a confirmation string be typed correctly
-   * before the submit button will be enabled.
-   */
   confirmationText: {
     type: String,
     default: '',
   },
-  preventProceedOnEnter: {
-    type: Boolean,
-    default: false,
+  confirmationPrompt: {
+    type: String,
+    default: 'Type {confirmationText} to confirm your action.',
   },
-  /**
-   * Options to be passed to tabbable
-   */
-  tabbableOptions: {
-    type: Object,
+  modalAttributes: {
+    type: Object as PropType<ModalAttributes>,
     default: () => ({}),
   },
 })
-const emit = defineEmits<{
-  (e: 'canceled'): void
-  (e: 'proceed', event: Event): void
-}>()
-
-const confirmationInput = ref('')
-
-const close = (): void => {
-  confirmationInput.value = ''
-  emit('canceled')
-}
-
-const proceed = (evt: Event): void => {
-  if (disableProceedButton.value) return
-
-  confirmationInput.value = ''
-  emit('proceed', evt)
-}
 
-const handleKeydown = (e: KeyboardEvent) => {
-  if (props.isVisible) {
-    if (e.key === 'Escape') {
-      close()
-    } else if (e.key === 'Enter') {
-      if (!props.preventProceedOnEnter) {
-        proceed(e)
-      }
-    }
-  }
-}
+const attrs = useAttrs()
 
-const capitalize = (str = ''): string => {
-  const capitalizeRegEx = /(?:^|[\s-:'"])\w/g
-  return str.replace(capitalizeRegEx, (a) => a.toUpperCase())
-}
+defineEmits<{
+  (e: 'cancel'): void
+  (e: 'proceed'): void
+}>()
 
-const displayTitle = computed((): string => {
-  if (props.title) {
-    if (props.type === 'warning') {
-      return 'Warning: ' + props.title
-    }
+const sanitizedAttrs = computed(() => {
+  const attributes = Object.assign({}, attrs) as Record<string, any>
 
-    return props.title
-  } else if (props.type === 'info') {
-    return 'Information'
-  }
+  // delete attributes that are handled through modalAttributes prop
+  delete attributes['tabbable-options']
+  delete attributes.width
+  delete attributes['max-height']
+  delete attributes['close-on-backdrop-click']
+  delete attributes['proceed-on-enter']
+  // delete KModal props that are not available in KPrompt
+  delete attributes['hide-close-icon']
+  delete attributes['hide-cancel-button']
```

Comment:
I question whether some of this is necessary. Props are not automatically bound to children the same way classes and styles are.

Reply:
Yes, it is necessary - I tested it and the prop is prop is propagated to KModal unless deleted like so (because KModal is not aa child of KPrompt - it's the first tag in template)